raiden/tests/unit/test_messages.py

```diff
@@ -143,3 +143,119 @@ def test_update_pfs():
+        tampered_balance_hash_request_monitoring.balance_proof.nonce,
+    )
+    # The signature works/is unaffected by that change...
+    assert recover(reward_proof_data, exploited_signature) == recover(
+        tampered_balance_hash_reward_proof_data,
+        exploited_signature,
+    )
+    assert recover(tampered_balance_hash_reward_proof_data, exploited_signature) == ADDRESS
+
+    # ...but overall verification fails
+    assert not tampered_balance_hash_request_monitoring.verify_request_monitoring(
+        PARTNER_ADDRESS,
+        ADDRESS,
+    )
+
+    # An attacker might change the additional_hash
+    partner_signed_balance_proof.additional_hash = 'tampered'.encode()
+
+    tampered_additional_hash_request_monitoring = RequestMonitoring(
+        onchain_balance_proof=partner_signed_balance_proof,
+        reward_amount=55,
+    )
+
+    tampered_additional_hash_reward_proof_data = pack_reward_proof(
+        tampered_additional_hash_request_monitoring.balance_proof.channel_identifier,
+        tampered_additional_hash_request_monitoring.reward_amount,
+        tampered_additional_hash_request_monitoring.balance_proof.token_network_address,
+        tampered_additional_hash_request_monitoring.balance_proof.chain_id,
+        tampered_additional_hash_request_monitoring.balance_proof.nonce,
+    )
+
+    # The signature works/is unaffected by that change...
+    assert recover(reward_proof_data, exploited_signature) == recover(
+        tampered_additional_hash_reward_proof_data,
+        exploited_signature,
+    )
+    assert recover(tampered_additional_hash_reward_proof_data,
+                   exploited_signature) == ADDRESS
+
+    # ...but overall verification fails
+    assert not tampered_balance_hash_request_monitoring.verify_request_monitoring(
+        PARTNER_ADDRESS,
+        ADDRESS,
+    )
+    # An attacker can change the non_closing_signature
+    partner_signed_balance_proof.non_closing_signature = 'tampered'.encode()
+
+    tampered_non_closing_signature_request_monitoring = RequestMonitoring(
+        onchain_balance_proof=partner_signed_balance_proof,
+        reward_amount=55,
+    )
+
+    tampered_non_closing_signature_reward_proof_data = pack_reward_proof(
+        tampered_non_closing_signature_request_monitoring.balance_proof.channel_identifier,
+        tampered_non_closing_signature_request_monitoring.reward_amount,
+        tampered_non_closing_signature_request_monitoring.balance_proof.token_network_address,
+        tampered_non_closing_signature_request_monitoring.balance_proof.chain_id,
+        tampered_non_closing_signature_request_monitoring.balance_proof.nonce,
+    )
+
+    # The signature works/is unaffected by that change...
+    assert recover(reward_proof_data, exploited_signature) == recover(
+        tampered_non_closing_signature_reward_proof_data,
+        exploited_signature,
+    )
+    assert recover(tampered_non_closing_signature_reward_proof_data, exploited_signature) == \
```

Comment:
please don't split lines with `\`. Instead use helper variables. It's more readable. e.g.:

```python
recovered_addrerss = recover(tampered_non_closing_signature_reward_proof_data, exploited_signature)
assert recovered == ADDRESS

Reply:
Done